mesonbuild/compilers/cuda.py

```diff
@@ -561,7 +561,7 @@ def sanity_check(self, work_dir: str, env: 'Environment') -> None:
             # a ton of compiler flags to differentiate between
             # arm and x86_64. So just compile.
             flags += self.get_compile_only_args()
-        flags += self.get_output_args(binary_name)
+        flags += self.get_output_args(binname)
```

Comment:
I don't understand why these are supposed to be wrong.

binname is `sanitycheckcuda`, binary name is `C:/path/to/builddir/meson-private/sanitycheckcuda.exe`. get_output_args returns its input with `-o` included.

This commit causes the cuda compiler to be responsible for appending ".exe"?

Also, we consistently use .exe for all sanity checks in other languages because it's still legal on Unix. :P

Reply:
I'm curious why passing absolute paths would fail here. It "shouldn't" matter either way, we set cwd=work_dir, so the change is conceptually fine to me, but you're saying it is actively broken before that...

Reply:
I don't think the language is harsh. The issue I had was that from reading it I didn't understand that the issue was the full paths at all. :)

(And also I can accept that it's total nonsense and shouldn't have ever worked to use full paths, but perhaps because I have no knowledge of cuda, I don't know *why* it's so terrible to pass them.)

Reply:
Well now I'm stumped. I remove the patch and everything works for me in both a cross compilation and a native compilation, but I was definitely running into it failing specifically because `$workdir/$bin` doesn't exist, but `$bin` does. I guess we'll push without the patch and see what happens in CI

Reply:
Okay, yeah, with that patch reverted run_project_tests.py doesn't detect Cuda and skips all of the tests in the cross-cuda case: https://github.com/mesonbuild/meson/actions/runs/16128151098/job/45509898038?pr=14758

Reply:
Okay, I've updated the commit message to be a little gentler, and to (attempt) to better describe what is being changed.

Reply:
Now I'm really curious, wanna add debug prints to make the CI report what's wrong without the patch?